neptune-core/src/application/database/storage/storage_schema/dbtmap_private.rs

```diff
@@ -0,0 +1,198 @@
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::fmt::Formatter;
+use std::hash::Hash;
+use std::sync::Arc;
+
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+use super::traits::StorageReader;
+use super::PendingWrites;
+use super::RustyKey;
+use super::RustyValue;
+use super::SimpleRustyReader;
+use super::WriteOperation;
+use crate::application::database::storage::storage_schema::DbtVec;
+use crate::application::database::storage::storage_vec::traits::StorageVecBase;
+use crate::application::database::storage::storage_vec::Index;
+use crate::application::locks::tokio::AtomicRw;
+
+/// A level-DB backed insertion-only mapping from keys to values.
+///
+/// The reason that remove functionality is not supported is that the
```

Comment:
I might be missing something, but at first thought it appears to me that DbtVec could efficiently support a delete or remove operation by adding a last_index property.  The delete op would simply delete the key at specified index and reduce the len.  This would leave a missing element, creating a sparse array.  Iteration would iter until last_index, skipping any elements not found.

Though rather than modify DbtVec itself, it may be better to create a new DbtVecSparse or something like that.

It's unclear to me how necessary a delete Op may be for DbtMap.   For this PR it is not necessary, but is it intended for general purpose usage, or just for this particular usage?

Reply:
The `DbtMap` is implemented for our concrete needs. I have not attempted to make it more generic than that. I have, however, attempted to follow principles that you have already established for database qualities, such as thread-safety although that is not strictly necessary for our use since it's all handled by the global state's rw-lock.